apiserver/facades/client/machinemanager/state.go

```diff
@@ -56,6 +58,8 @@ type Machine interface {
 	WatchUpgradeSeriesNotifications() (state.NotifyWatcher, error)
 	GetUpgradeSeriesMessages() ([]string, bool, error)
 	IsManager() bool
+	IsLockedForSeriesUpgrade() (bool, error)
```

Comment:
Something that bugs me about this implementation in state is that the methods are on `Machine`, but (mostly?) only touch the collection for series-upgrade locks.

It would be nice to deal with this behaviour without materialising a machine if we don't need it, but that's a consideration for another day.

Reply:
:rabbit: :hole: 

If I didn't have bigger fish to fry I'd do that as well!